packages/e2e-tests/specs/undo.test.js

```diff
@@ -54,27 +54,49 @@ describe( 'undo', () => {
 		expect( await getEditedPostContent() ).toBe( '' );
 	} );
 
-	it( 'Should undo to expected level intervals', async () => {
+	it( 'Should undo/redo to expected level intervals', async () => {
 		await clickBlockAppender();
 
 		await page.keyboard.type( 'This' );
 		await page.keyboard.press( 'Enter' );
 		await page.keyboard.type( 'is' );
 		await page.keyboard.press( 'Enter' );
 		await page.keyboard.type( 'test' );
+		await new Promise( ( resolve ) => setTimeout( resolve, 1000 ) );
```

Comment:
Why is this timeout added? Can't we add a separate test case with the timeout? Are we still testing without the timeout if we add it here?

Reply:
The test fail when this is removed, which is curious. This line shouldn't be necessary for it to pass.

Reply:
We need it to make sure the timeout after typing for the undo level creation fires.

Reply:
Hm, this shouldn't be necessary. You should be able to undo and redo without waiting a second.

Reply:
How does the last level get created otherwise? It's implemented with a timeout.

Reply:
This is something that was previously tested with `withHistory` unit tests. Do you think it's possible to add unit tests for this new reducer?

Reply:
Yes

Reply:
https://github.com/WordPress/gutenberg/pull/17861/commits/ce90a828cff583b645596cd4b7ee3493741b0468